src/compiler/utilities.ts

```diff
@@ -1637,6 +1637,9 @@ namespace ts {
                 const rightHandSideName = (<PropertyAccessExpression>nameExpression).name.text;
                 return getPropertyNameForKnownSymbolName(rightHandSideName);
             }
+            if (isStringOrNumericLiteral(nameExpression.kind)) {

```

Comment:
Does this fix the problem we saw on https://github.com/Microsoft/TypeScript/pull/7814/files#r58450947?